### ISSUE TITLE
[Feature Request] Verified Purchase

### ISSUE THREAD

**Issue by 23d1** [owldesign/QARR#22]
Not sure if it's worth it, but perhaps by checking the email associated with a submission against past orders emails could flip a switch. Or, is it possible to have a hidden field in a Display to incorporate this as a switch in the backend (for manual verification)?

**Comment by owldesign** [comment]
@23d1 I think I can check submission email against commerce orders and check if product being reviewed is in that order. And you just want some type of indication for each review/question submission that shows whether there is a match?

**Comment by 23d1** [comment]
Actually, the more I think about it, it would be sufficient to just be able to have extra fields in the backend on submission, like Displays. I haven't gotten around to test yet, but does that work currently, setting up a Display, then having field included that are not rendered in the client templates?

**Comment by owldesign** [comment]
Currently all fields within Displays will render on the front end form

**Comment by 23d1** [comment]
Even with custom form template?

**Comment by owldesign** [comment]
Yeah you're right, you can add a hidden field to custom forms (https://docs.qarr.tools/custom/reviews/#reviews). But how are you thinking about it? Did you want to hit some function that checks if verified customer? 
And after submittion this hidden field needs to show up in your entry submisison page?

**Comment by 23d1** [comment]
Well, at first I thought it might be good to have something in QARR that checks if the user posting a review or question on a product has supplied an email for their post that is found on an order, and checking against the lineitems on the order if they have at some point purchased the product they are reviewing/asking a question about. With the ability to also call this out in client facing templates as a simple true/false.

**Comment by owldesign** [comment]
So I might have already thought about this direction but never implemented. I have a column called `hasPurchased` in db for reviews and questions. 

We can programatically check for that after submission or I guess technically if you add a field to Display (checkbox) with `name="fields[hasPurchased]` so users can check if they purchased product before, but then yeah you'd have to manually verify if true.

**Comment by 23d1** [comment]
Oh, awesome. Yeah—if there's an automated approach, that's definitely preferred. If sales volumes are high then it would get crazy pretty quick to do it manually. :)

**Comment by owldesign** [comment]
Already done! I'm making other updates to the plugin, so you'll have it on the next push.

**Comment by 23d1** [comment]
Right on! Quick question; what happens when there are multiple orders? Would it simply limit to the latest one (makes most sense I think, first or last).

**Comment by owldesign** [comment]
Right now all im doing is just getting `.one()` result. So whatever Craft thinks `.one()` on an element query is (the most recent?) that's what is being returned. 

I can also set it to `.all()` and get all orders this customer had with this product. 

Everything is in place so whatever the use case makes more sense I'm fine with it.

**Comment by 23d1** [comment]
Gotcha. I think `.one()` is fine, I'm a big fan of keeping it simple. There would probably be completely other use cases in terms of email marketing and things, but that's outside of this plugins' scope, no doubt, as the intent is to get people who ordered to review, which is then verified against actual orders.

**Comment by owldesign** [comment]
Ok. 

I’m also working on Campaign functionality for the plugin. So if you got ideas/suggestions for email marking portion that’ll be awesome help. Being able to mass email people who’ve purchased a product to review will probably be my starting point.

**Comment by 23d1** [comment]
Awesome!
One thing that would be great is the ability to time emails like say after 14 or 21 days send email to request rating/review for the items in the order. The link could then either be the product page, or if one has a custom page set up that just houses the form for posting rating/review for each specific product (seems like the least amount of steps, not having to locate the review functionality on the product page).
Not sure if timing would require a cron job, but if so, would be good to be able to use an external cron service to simply hit a URL with a "secret" hash or something like that like https://url.com/qarr?W34RfgfggeiuUg if that makes sense? :)
